Tests/PackageCollectionsTests/JSONPackageCollectionProviderTests.swift

```diff
@@ -269,7 +269,12 @@ class JSONPackageCollectionProviderTests: XCTestCase {
         httpClient.configuration.retryStrategy = .none
         let provider = JSONPackageCollectionProvider(httpClient: httpClient, diagnosticsEngine: DiagnosticsEngine())
         XCTAssertThrowsError(try tsc_await { callback in provider.get(source, callback: callback) }, "expected error", { error in
-            XCTAssertEqual(error as? HTTPClientError, .badResponseStatusCode(statusCode))
+            switch error {
+            case JSONPackageCollectionProvider.Errors.collectionUnavailable(_, let status):
+                XCTAssertEqual(status, statusCode)
+            default:
+                XCTFail("unexpected error \(error)")
+            }
```

Comment:
can this work?: `XCTAssertEqual(error as? JSONPackageCollectionProvider.Error, .collectionUnavailable(url , statusCode))`

Reply:
Had to change `JSONPackageCollectionProvider.Errors` a little in order to conform it to `Equatable`